calour/training.py

```diff
@@ -14,11 +14,16 @@
    add_sample_metadata_as_features
 '''
 
-
 from logging import getLogger
+import itertools
 
 from sklearn.feature_extraction import DictVectorizer
-from sklearn.model_selection import train_test_split
+from sklearn.model_selection import train_test_split, RepeatedStratifiedKFold, ParameterGrid
+from sklearn.model_selection._split import check_cv
+from sklearn.base import is_classifier
+from sklearn.metrics import roc_curve, auc, confusion_matrix
+from matplotlib import pyplot as plt
```

Comment:
i think it's a problem to import pyplot here since then pyplot is imported when the calour is loaded, and then it sets the matplotlib backend.
we should import in the function that uses pyplot instead.

Reply:
I remember this - but why its setting backend is a problem? as long as it doesn't change the backend, it's fine?

Reply:
i think the problem is with the qt5 gui vs. the notebook gui... don't remember exactly...
anyway, best to move it into the functions that need it :)